pkg/framework/plugins/removepodsviolatinginterpodantiaffinity/pod_antiaffinity.go

```diff
@@ -14,80 +14,88 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package strategies
+package removepodsviolatinginterpodantiaffinity
 
 import (
 	"context"
 	"fmt"
 
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/sets"
-	"sigs.k8s.io/descheduler/pkg/api"
+	"sigs.k8s.io/descheduler/pkg/apis/componentconfig"
 	"sigs.k8s.io/descheduler/pkg/descheduler/evictions"
 	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+	"sigs.k8s.io/descheduler/pkg/framework"
 	"sigs.k8s.io/descheduler/pkg/utils"
 
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/klog/v2"
 )
 
-func validateRemovePodsViolatingInterPodAntiAffinityParams(params *api.StrategyParameters) error {
-	if params == nil {
-		return nil
-	}
-
-	// At most one of include/exclude can be set
-	if params.Namespaces != nil && len(params.Namespaces.Include) > 0 && len(params.Namespaces.Exclude) > 0 {
-		return fmt.Errorf("only one of Include/Exclude namespaces can be set")
-	}
-	if params.ThresholdPriority != nil && params.ThresholdPriorityClassName != "" {
-		return fmt.Errorf("only one of thresholdPriority and thresholdPriorityClassName can be set")
-	}
+const PluginName = "RemovePodsViolatingInterPodAntiAffinity"
 
-	return nil
+// RemovePodsViolatingInterPodAntiAffinity evicts pods on the node which violate inter pod anti affinity
+type RemovePodsViolatingInterPodAntiAffinity struct {
+	handle    framework.Handle
+	args      *componentconfig.RemovePodsViolatingInterPodAntiAffinityArgs
+	podFilter podutil.FilterFunc
 }
 
-// RemovePodsViolatingInterPodAntiAffinity evicts pods on the node which are having a pod affinity rules.
-func RemovePodsViolatingInterPodAntiAffinity(ctx context.Context, client clientset.Interface, strategy api.DeschedulerStrategy, nodes []*v1.Node, podEvictor *evictions.PodEvictor, evictorFilter *evictions.EvictorFilter, getPodsAssignedToNode podutil.GetPodsAssignedToNodeFunc) {
-	if err := validateRemovePodsViolatingInterPodAntiAffinityParams(strategy.Params); err != nil {
-		klog.ErrorS(err, "Invalid RemovePodsViolatingInterPodAntiAffinity parameters")
-		return
+var _ framework.Plugin = &RemovePodsViolatingInterPodAntiAffinity{}
+var _ framework.DeschedulePlugin = &RemovePodsViolatingInterPodAntiAffinity{}
```

Comment:
I'm questioning whether this should have been a Balance plugin in the proposal.

By [definition](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/): "Inter-pod affinity/anti-affinity allows you to constrain Pods against labels on other Pods."

Reply:
The strategy does not perform any pod balancing at the moment.
```go
if checkPodsWithAntiAffinityExist(pods[i], pods) && d.handle.Evictor().Filter(pods[i]) {
	if d.handle.Evictor().Evict(ctx, pods[i], evictions.EvictOptions{}) {
		// Since the current pod is evicted all other pods which have anti-affinity with this
		// pod need not be evicted.
		// Update pods.
		pods = append(pods[:i], pods[i+1:]...)
		i--
		totalPods--
	}
}
```
might be extended to store pods into anti-affinity groups. Then during eviction remove pods from groups which have the most pods first to reduce size of groups through out all namespaces as a balancing metric. No objections for implementing such functionality and changing the extension point then.

Reply:
In a followup PR you mean? We can raise an issue to track it

Reply:
@knelasevero yes, this is an RFE for a new PR. +1 for tracking it in a new issue.

Reply:
Issue at https://github.com/kubernetes-sigs/descheduler/issues/914